src/AddTodoForm.jsx

```diff
@@ -3,19 +3,25 @@ import { useState } from "react";
 import InputWithLabel from "./InputWithLabel";
 
 const AddTodoForm = (props) => {
-  const { onAddTodo } = props;
+  const { onAddTodo, onRemoveTodo, onPostData } = props;
   const [todoTitle, setTodoTitle] = useState("");
 
   const handleTitleChange = (event) => {
     const newTodoTitle = event.target.value;
     setTodoTitle(newTodoTitle);
   };
 
-  const handleAddTodo = (event) => {
+  const handleAddTodo = async (event) => {
     event.preventDefault();
-    console.log(todoTitle);
-    setTodoTitle("");
-    onAddTodo({ title: todoTitle, id: Date.now() });
+    try {
+      onAddTodo({ title: todoTitle, id: null });
+      const taskId = await onPostData(todoTitle);
+      onRemoveTodo(null);
+      onAddTodo({ title: todoTitle, id: taskId });
+      setTodoTitle("");
```

Comment:
You should not do this because you are calling onAddTodoTwice, what happens if there is an error

this is what you should do 
- post the data first 
- when it returns the id then call onAddTodo with the id it returns

```suggestion
      const newTodo = await onPostData(todoTitle);
      onAddTodo(newTodo);
      setTodoTitle("");
```

Then instead of returning just the todoId from the post request, return the whole todo object